src/environment.jl

```diff
@@ -158,3 +158,28 @@ function Wtime()
     end
 end
 
+
+"""
+    MPI.has_cuda()
+
+Check if the MPI implementation is known to have CUDA support. Currently only Open MPI
+provides a mechanism to check, so it will return `false` with other implementations
+(unless overriden).
+
+This can be overriden by setting the `JULIA_MPI_HAS_CUDA` environment variable to `true`
+or `false`.
+"""
+function has_cuda()
+    flag = get(ENV, "JULIA_MPI_HAS_CUDA", nothing)
+    if flag === nothing
```

Comment:
Could we use this to actually provide a function `MPI_vendor()`?

@leios wanted something like that.

Reply:
My plan is to do this as part of adding BinaryProvider support, but I don’t have the bandwidth to do this now

Reply:
#354 does that but doesn't help with CUDA awareness.

Reply:
I've switched this to use #354.